File: src/RoadApproach.cpp
#include "RoadApproach.h"

RoadApproach::RoadApproach(std::string direction)
    : directionId(std::move(direction)), trafficLight(directionId), vehicleSensor(directionId) {}

TrafficLight &RoadApproach::light() { return trafficLight; }

Sensor &RoadApproach::sensor() { return vehicleSensor; }

const TrafficLight &RoadApproach::light() const { return trafficLight; }

const Sensor &RoadApproach::sensor() const { return vehicleSensor; }

const std::string &RoadApproach::direction() const { return directionId; }

File: python/src/road_approach.py
from .traffic_light import TrafficLight
from .sensor import Sensor

class RoadApproach:
    def __init__(self, direction: str):
        self._direction = direction
        self._light = TrafficLight(direction)
        self._sensor = Sensor()

    def direction(self) -> str:
        return self._direction

    def light(self) -> TrafficLight:
        return self._light

    def sensor(self) -> Sensor:
        return self._sensor
